app/components/meeting/edit-meeting-modal.js

```diff
@@ -222,41 +222,61 @@ export default class MeetingEditMeetingComponent extends Component {
       );
   }
 
+  regenerateDecisionReport = task(async (decisionActivity) => {
+    if (this.enableDigitalAgenda) {
+      const report = await this.store.queryOne('report', {
+        'filter[:has-no:next-piece]': true,
+        'filter[decision-activity][:id:]': decisionActivity.id,
+      });
+      const pieceParts = await report?.pieceParts;
+      if (pieceParts?.length) {
+        await this.decisionReportGeneration.generateReplacementReport.perform(
+          report
+        );
+      }
+    }
+  });
+
   @task
   *saveMeeting() {
     const now = new Date();
 
-    this.args.meeting.extraInfo = this.extraInfo;
-    this.args.meeting.plannedStart = this.startDate || now;
-    this.args.meeting.kind = this.selectedKind;
-    this.args.meeting.number = this.meetingNumber;
-    this.args.meeting.numberRepresentation = this.numberRepresentation;
-    this.args.meeting.mainMeeting = this.selectedMainMeeting;
-
     if (this.enableDigitalAgenda) {
       const currentMeetingSecretary = yield this.args.meeting.secretary;
-      if (currentMeetingSecretary?.uri !== this.secretary.uri) {
+      const currentKind = yield this.args.meeting.kind;
+      const currentPlannedStart = this.args.meeting.plannedStart;
+      if (
+        currentMeetingSecretary?.uri !== this.secretary.uri ||
+        currentKind?.uri !== this.selectedKind.uri ||
+        currentPlannedStart !== this.startDate || 
+        currentPlannedStart !== now 
+      ) {
+        this.args.meeting.kind = this.selectedKind;
         this.args.meeting.secretary = this.secretary;
-        const decisionActivities = yield this.store.queryAll('decision-activity', {
-          'filter[treatment][agendaitems][agenda][created-for][:id:]':
-            this.args.meeting.id,
-        });
+        this.args.meeting.plannedStart = this.startDate || now;
+        yield this.args.meeting.save();
```

Comment:
**question (non-blocking)**: any reason we don't move this PDF re-generation logic after we save the meeting?

Reply:
Yeah makes sense, I was being lazy because moving the code up allowed me to reuse existing variables. Will change

Reply:
fixed in [9553991](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1905/commits/95539913c4dd18d607bf4320e3980fa69a130273)